pointillism-port.patch

```diff
@@ -0,0 +1,116 @@
+diff --git a/activity.py b/activity.py
+index d91ea44..75d69de 100755
+--- a/activity.py
++++ b/activity.py
+@@ -24,14 +24,16 @@
+ 
+ import puntillism
+ from gettext import gettext as _
+-from sugar3.datastore import datastore
+ from sugar3.graphics.objectchooser import ObjectChooser
+ from sugar3.graphics.toolbutton import ToolButton
+ from sugar3.activity.widgets import StopButton
+ from sugar3.activity.widgets import ActivityToolbarButton
+ from sugar3.graphics.toolbarbox import ToolbarBox
+-from sugar3.activity import activity
+-from sugar3 import mime
++import shutil
++import sys
++from sugarapp.widgets import SugarCompatibleActivity
++from sugarapp.widgets import DesktopSaveChooser
++from sugarapp.widgets import DesktopOpenChooser
+ import sugargame.canvas
+ import sugargame
+ import pygame
+@@ -43,10 +45,10 @@ gi.require_version('Gdk', '3.0')
+ gi.require_version('Gtk', '3.0')
+ 
+ 
+-class Activity(activity.Activity):
++class Activity(SugarCompatibleActivity):
+ 
+     def __init__(self, handle):
+-        activity.Activity.__init__(self, handle)
++        SugarCompatibleActivity.__init__(self, handle)
+ 
+         self.max_participants = 1
+         self.file_path_temp = None
+@@ -56,13 +58,11 @@ class Activity(activity.Activity):
+         self.actividad = puntillism.Puntillism(self)
+ 
+         self.build_toolbar()
+-
+         self._pygamecanvas = \
+             sugargame.canvas.PygameCanvas(
+                 self,
+                 main=self.actividad.run,
+                 modules=[pygame.display])
+-
+         self.set_canvas(self._pygamecanvas)
+         self._pygamecanvas.grab_focus()
+ 
+@@ -155,19 +155,18 @@ class Activity(activity.Activity):
+     def _savebutton_cb(self, button):
+         pygame.event.post(pygame.event.Event(
+             pygame.USEREVENT, action='savebutton'))
+-
++    
++    def can_close(self):
++        self.actividad.running = False
++        self.load_image_loop = False
++        pygame.quit()
```

Comment:
The version I reviewed early today did not have the quit nor the exit line, and worked just fine. Why adding it back?